### PR TITLE
Fix OperationalError: too many SQL variables

### PR DESCRIPTION
**Pull request by fernandog**
Fixes https://github.com/pymedusa/Medusa/issues/1716

### CODE REVIEW

**Comment by fernandog** [comment]
@labrys you reviewed an old merged PR about popsm is that right?

**Comment by fernandog** [comment]
I looked all queries that somehow uses a join of "?" and I don't see any other that should use this helper

**Comment by p0psicles** [comment]
What's bugging me in this PR, is that we are adding a work-around, for a situation that should never occur right? Why would we want to delete a very large list of networks?

**Comment by fernandog** [comment]
@p0psicles i see no errors while getting the networks in his log

what code basic does is load all cached networks to network_list var,
if the remote network is found in cached, we remove from the network_list var
so in the end, we delete all networks from network_list

what we can do is: if we read 0 networks from remote, don't do anything to the cached one

**Comment by fernandog** [comment]
what is wrong is this:
```
    except (IOError, OSError):
        pass
```
maybe some ERROR and the variable that keeps the remove network is empty, so it delete it all

**Comment by fernandog** [comment]
now i abort if exception. I see no other reason that could cause a massive delete.
thanks for the feedback

File: medusa/network_timezones.py [hunk 1]
```diff
@@ -70,14 +73,17 @@ def update_network_dict():
...
+            queries.append(['UPDATE OR IGNORE network_timezones '
+                            'SET timezone = ? WHERE network_name = ?;', [timezone, network]])
 
         if existing:
             del network_list[network]
 
     if network_list:
         purged = [x for x in network_list]
-        queries.append(['DELETE FROM network_timezones WHERE network_name IN (%s);' % ','.join(['?'] * len(purged)), purged])
+        for paged_list in get_rows_by_slice(purged, 50):
```
Comment (p0psicles) [inline medusa/network_timezones.py]: This can be reverted.

File: medusa/helpers.py [hunk 2]
```diff
@@ -1729,3 +1729,9 @@ def is_info_hash_processed(info_hash):
                                         'WHERE d.action LIKE "%04"',
                                         [info_hash])
     return bool(history_result)
+
+
+def get_rows_by_slice(seq, row_len):
```
Comment (p0psicles) [inline medusa/helpers.py]: As this.

File: medusa/network_timezones.py [hunk 3]
```diff
@@ -25,6 +25,7 @@
 from six import iteritems
 from . import db, helpers, logger
 from .helper.common import try_int
+from .helpers import get_rows_by_slice
```
Comment (p0psicles) [inline medusa/network_timezones.py]: As this.

File: medusa/network_timezones.py (hunk 1 above)
Comment (fernandog) [inline medusa/network_timezones.py]: @p0psicles Why not keep for safety?

**Comment by fernandog** [comment]
@p0psicles waiting approval as its ready

**Pull request merged**